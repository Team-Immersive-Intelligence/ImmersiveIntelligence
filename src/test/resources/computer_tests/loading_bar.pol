;Creates an animated progress bar

use ADVANCED_ARITHMETIC ;use a circuit
use COMPARATOR
use LOGIC
use TYPE_CONVERSION
use TEXT

integer w = 5  ;wait time
integer m = 64 ;max
integer i = 0   ;begin
integer l = 32  ;length (in chars)

mark loop

string b = "["
float k = * @l (/ (<float> @i) (<float> @m)) ;current length
float f = * (/ (<float> @i) (<float> @m)) 100
copy k e; empty chars
float e = - @l @e

mark full
if > @k 0
    float k = - @k 1
    string b = string_join @b "="
    goto "full"

mark empty
if > @e 0
    float e = - @e 1
    string b = string_join @b " "
    goto "empty"

string b = string_join @b "]"
string b = string_join @b " "
string b = string_join @b (<string> @f)
string b = string_join @b "%"

wait @w ;wait w ticks
type @b
integer i = + @i 1

if < @i @m
    goto "loop"

type "Loading Complete!"

end