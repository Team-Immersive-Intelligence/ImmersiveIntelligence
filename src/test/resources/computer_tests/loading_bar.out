[POL] [                                ] 0.0%
[POL] [=                                ] 1.5625%
[POL] [=                               ] 3.125%
[POL] [==                               ] 4.6875%
[POL] [==                              ] 6.25%
[POL] [===                              ] 7.8125%
[POL] [===                             ] 9.375%
[POL] [====                             ] 10.9375%
[POL] [====                            ] 12.5%
[POL] [=====                            ] 14.0625%
[POL] [=====                           ] 15.625%
[POL] [======                           ] 17.1875%
[POL] [======                          ] 18.75%
[POL] [=======                          ] 20.3125%
[POL] [=======                         ] 21.875%
[POL] [========                         ] 23.4375%
[POL] [========                        ] 25.0%
[POL] [=========                        ] 26.5625%
[POL] [=========                       ] 28.125%
[POL] [==========                       ] 29.6875%
[POL] [==========                      ] 31.25%
[POL] [===========                      ] 32.8125%
[POL] [===========                     ] 34.375%
[POL] [============                     ] 35.9375%
[POL] [============                    ] 37.5%
[POL] [=============                    ] 39.0625%
[POL] [=============                   ] 40.625%
[POL] [==============                   ] 42.1875%
[POL] [==============                  ] 43.75%
[POL] [===============                  ] 45.3125%
[POL] [===============                 ] 46.875%
[POL] [================                 ] 48.4375%
[POL] [================                ] 50.0%
[POL] [=================                ] 51.5625%
[POL] [=================               ] 53.125%
[POL] [==================               ] 54.6875%
[POL] [==================              ] 56.25%
[POL] [===================              ] 57.8125%
[POL] [===================             ] 59.375%
[POL] [====================             ] 60.9375%
[POL] [====================            ] 62.5%
[POL] [=====================            ] 64.0625%
[POL] [=====================           ] 65.625%
[POL] [======================           ] 67.1875%
[POL] [======================          ] 68.75%
[POL] [=======================          ] 70.3125%
[POL] [=======================         ] 71.875%
[POL] [========================         ] 73.4375%
[POL] [========================        ] 75.0%
[POL] [=========================        ] 76.5625%
[POL] [=========================       ] 78.125%
[POL] [==========================       ] 79.6875%
[POL] [==========================      ] 81.25%
[POL] [===========================      ] 82.8125%
[POL] [===========================     ] 84.375%
[POL] [============================     ] 85.9375%
[POL] [============================    ] 87.5%
[POL] [=============================    ] 89.0625%
[POL] [=============================   ] 90.625%
[POL] [==============================   ] 92.1875%
[POL] [==============================  ] 93.75%
[POL] [===============================  ] 95.3125%
[POL] [=============================== ] 96.875%
[POL] [================================ ] 98.4375%
[POL] Loading Complete!
